show_results() not working when called on a fp16 Learner

**Description**
`show_results()` crashes when called on a half-precision  `Learner`.


**Installation details**
```text
=== Software === 
python        : 3.7.1
fastai        : 1.0.42
fastprogress  : 0.1.18
torch         : 1.0.1.post2
nvidia driver : 410.78
torch cuda    : 10.0.130 / is available
torch cudnn   : 7402 / is enabled

=== Hardware === 
nvidia gpus   : 1
torch devices : 1
  - gpu0      : 7929MB | GeForce RTX 2070

=== Environment === 
platform      : Linux-4.15.0-45-generic-x86_64-with-debian-buster-sid
distro        : #48-Ubuntu SMP Tue Jan 29 16:28:13 UTC 2019
conda env     : fastai
python        : /home/weenkus/anaconda3/envs/fastai/bin/python
sys.path      : /home/weenkus/Workspace/Training/course-v3/nbs/dl1
/home/weenkus/anaconda3/envs/fastai/lib/python37.zip
/home/weenkus/anaconda3/envs/fastai/lib/python3.7
/home/weenkus/anaconda3/envs/fastai/lib/python3.7/lib-dynload
/home/weenkus/anaconda3/envs/fastai/lib/python3.7/site-packages
/home/weenkus/anaconda3/envs/fastai/lib/python3.7/site-packages/IPython/extensions
```

**To Reproduce**
I managed to reproduce it by adding these tests to `test_vision_train.py`;

```python
# PASSES
def test_show_results(learn):
    learn.show_results()

# PASSES
def test_show_results_16bit_to_32bit(learn):
    learn.to_fp16()
    learn.to_fp32()
    learn.show_results()

# CRASHES
def test_show_results_32bit(learn):
    learn.to_fp16()
    learn.show_results()
```

The first two tests pass with 32bit models while the third tests crashes because of the half-precision model.

**Expected behavior**
I expected a half-precision learner after training to be able to call the method `show_results()`

**Screenshots**
Reproduced it via the lesson 1 (part 1 2019 course) jupyter notebook.
![Screenshot image](https://i.imgur.com/W5Z09iC.png)

Code;
```python
# CELL 1
%reload_ext autoreload
%autoreload 2
%matplotlib inline

from fastai.vision import *
from fastai.metrics import error_rate

path = untar_data(URLs.PETS)

path_anno = path/'annotations'
path_img = path/'images'
fnames = get_image_files(path_img)
pat = r'/([^/]+)_\d+.jpg$'

data = ImageDataBunch.from_name_re(
    path_img, fnames, pat, ds_tfms=get_transforms(), size=224, bs=64
).normalize(imagenet_stats)

# CELL 2
learn = create_cnn(data, models.resnet34, metrics=error_rate)
learn.fit_one_cycle(1)
learn.show_results(rows=1)

# CELL 3
learn = create_cnn(data, models.resnet34, metrics=error_rate)
learn.to_fp16()
learn.fit_one_cycle(1)
learn.show_results(rows=1)
```

@sgugger one of these tests are still failing for me (even with the fix PR: https://github.com/fastai/fastai/commit/3985dc8e4ae86ea800df93193f1d1c46ad8e8c48)

In `tests/test_vision_training.py`
```python

def test_show_results(learn):
    learn.show_results()

def test_show_results_16bit_to_32bit(learn):
    learn.to_fp16()
    learn.to_fp32()
    learn.show_results()

def test_show_results_32bit(learn):
    learn.to_fp16()
    learn.show_results()
```

Maybe we should add these tests as well.

Error:
```
fastai/vision/learner.py:102: in _cl_int_from_learner
    preds = learn.TTA(ds_type=ds_type,with_loss=True) if tta else learn.get_preds(ds_type=ds_type, with_loss=True)
fastai/basic_train.py:278: in get_preds
    activ=_loss_func2activ(self.loss_func), loss_func=lf, n_batch=n_batch, pbar=pbar)
fastai/basic_train.py:40: in get_preds
    zip(*validate(model, dl, cb_handler=cb_handler, pbar=pbar, average=False, n_batch=n_batch))]
fastai/basic_train.py:54: in validate
    val_losses.append(loss_batch(model, xb, yb, loss_func, cb_handler=cb_handler))
fastai/basic_train.py:20: in loss_batch
    out = model(*xb)
../../../anaconda3/envs/fastai_dev/lib/python3.7/site-packages/torch/nn/modules/module.py:489: in __call__
    result = self.forward(*input, **kwargs)
../../../anaconda3/envs/fastai_dev/lib/python3.7/site-packages/torch/nn/modules/container.py:92: in forward
    input = module(input)
../../../anaconda3/envs/fastai_dev/lib/python3.7/site-packages/torch/nn/modules/module.py:489: in __call__
    result = self.forward(*input, **kwargs)
../../../anaconda3/envs/fastai_dev/lib/python3.7/site-packages/torch/nn/modules/container.py:92: in forward
    input = module(input)
../../../anaconda3/envs/fastai_dev/lib/python3.7/site-packages/torch/nn/modules/module.py:489: in __call__
    result = self.forward(*input, **kwargs)

self = Conv2d(3, 16, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1), bias=False)
input = tensor([[[[-0.4000, -0.4000, -0.4000,  ..., -0.4000, -0.4000, -0.4000],
          [-0.4000, -0.4000, -0.4000,  ..., -0..., -0.4000, -0.4000],
          [-0.4000, -0.4000, -0.4000,  ..., -0.4000, -0.4000, -0.4000]]]],
       device='cuda:0')

    @weak_script_method
    def forward(self, input):
        return F.conv2d(input, self.weight, self.bias, self.stride,
>                       self.padding, self.dilation, self.groups)
E       RuntimeError: Input type (torch.cuda.FloatTensor) and weight type (torch.cuda.HalfTensor) should be the same

../../../anaconda3/envs/fastai_dev/lib/python3.7/site-packages/torch/nn/modules/conv.py:320: RuntimeError
```

Seems it has to do something with `loss_batch()` being called with weights being in `half()` while the input being `float()`.

--- a/fastai/vision/data.py
+++ b/fastai/vision/data.py
@@ -58,10 +58,7 @@ def normalize(x:TensorImage, mean:FloatTensor,std:FloatTensor)->TensorImage:
 
 def denormalize(x:TensorImage, mean:FloatTensor,std:FloatTensor, do_x:bool=True)->TensorImage:
     "Denormalize `x` with `mean` and `std`."
-    if x.dtype == torch.float16:
-        return x.cpu().float()*std[...,None,None] + mean[...,None,None] if do_x else x.cpu()
-    else:
-        return x.cpu()*std[...,None,None] + mean[...,None,None] if do_x else x.cpu()
+    return x.cpu().float()*std[...,None,None] + mean[...,None,None] if do_x else x.cpu()
 
 def _normalize_batch(b:Tuple[Tensor,Tensor], mean:FloatTensor, std:FloatTensor, do_x:bool=True, do_y:bool=False)->Tuple[Tensor,Tensor]:
     "`b` = `x`,`y` - normalize `x` array of imgs and `do_y` optionally `y`."

--- a/fastai/vision/data.py
+++ b/fastai/vision/data.py
@@ -58,7 +58,10 @@ def normalize(x:TensorImage, mean:FloatTensor,std:FloatTensor)->TensorImage:
 
 def denormalize(x:TensorImage, mean:FloatTensor,std:FloatTensor, do_x:bool=True)->TensorImage:
     "Denormalize `x` with `mean` and `std`."
-    return x.cpu()*std[...,None,None] + mean[...,None,None] if do_x else x.cpu()
+    if x.dtype == torch.float16:
+        return x.cpu().float()*std[...,None,None] + mean[...,None,None] if do_x else x.cpu()
+    else:
+        return x.cpu()*std[...,None,None] + mean[...,None,None] if do_x else x.cpu()
 
 def _normalize_batch(b:Tuple[Tensor,Tensor], mean:FloatTensor, std:FloatTensor, do_x:bool=True, do_y:bool=False)->Tuple[Tensor,Tensor]:
     "`b` = `x`,`y` - normalize `x` array of imgs and `do_y` optionally `y`."

